### DISCUSSION TITLE
WebView2 Read Website in System.Windows.Forms.HtmlDocument variable

### DISCUSSION THREAD

**Discussion by Michael091971** [MicrosoftEdge/WebView2Feedback#2492] [Q&A]
In the Webbrowser Control there is the possibility to save the Website in the System.Windows.Forms.HtmlDocument variable. After that it is possible to search in the html-Variable. What is the function in WebView2 that makes the same? 

  Private Sub WebBrowser1_Navigated(ByVal sender As Object, ByVal e As System.Windows.Forms.WebBrowserNavigatedEventArgs) Handles WebBrowser1.Navigated
Dim document As System.Windows.Forms.HtmlDocument = WebBrowser1.Document
cUrl_Kundendisplay = document.All("Kundendisplaylink").GetAttribute("value")   ' Search for the HTML-Var "Kundendisplaylink"

 Private Async Sub WebView_NavigationCompleted(sender As Object, e As CoreWebView2NavigationCompletedEventArgs) Handles WebView.NavigationCompleted
Dim document As System.Windows.Forms.HtmlDocument
cUrl_Kundendisplay = ???
Kundendisplay.WebView2_2.Source = New Uri(cUrl_Kundendisplay, UriKind.Absolute)

**Comment by amaitland** [comment]
See https://github.com/MicrosoftEdge/WebView2Feedback/issues/176#issuecomment-629358865 for an official response. 

You can use JavaScript or one of third party DOM libraries. See 
 https://github.com/MicrosoftEdge/WebView2Feedback/issues/77#issuecomment-1073677281 for details on my DevTools based library.
